scc/light_client/light_client.go

```diff
@@ -64,3 +67,78 @@ func (c *LightClient) Close() {
 func (c *LightClient) Sync() (idx.Block, error) {
 	return c.state.sync(c.provider)
 }
+
+// getAccountProof retrieves and verifies the proof for the given address.
+// It first ensures the client is synchronized before querying for the proof.
+// Returns an error if synchronization fails, if the proof cannot be obtained.
+func (c *LightClient) getAccountProof(address common.Address) (carmen.WitnessProof, error) {
+	// always sync before querying
+	_, err := c.Sync()
+	if err != nil {
+		return nil, fmt.Errorf("failed to sync: %w", err)
+	}
+	proof, err := c.provider.getAccountProof(address, LatestBlock)
```

Comment:
This may simply fail because "LatestBlock" has moved since the completion of the `Sync` call. In fact, it is quite likely that the certification chain is lagging behind the change by a few blocks. So this fetch should not ask for the "LatestBlock" but the block you just synced to.

Reply:
right. Here it should use the latest block known to the light client state.